prime/src/data/BalanceSheet.ts

```diff
@@ -300,8 +302,10 @@ export function maxWithdraws(
   if (surplus1C.isGteZero()) {
     if (surplus0C.isLteZero()) {
       denom0 = priceB.recklessMul(coeff).add(priceC.recklessDiv(ALOE_II_LIQUIDATION_INCENTIVE));
-    } else if (surplus0C.lt(maxWithdrawB0.div(priceB))) {
-      maxWithdrawB0 = maxWithdrawB0.add(surplus0C.mul(priceC).recklessDiv(ALOE_II_LIQUIDATION_INCENTIVE));
+    } else if (surplus0C.lt(maxWithdrawB0.setResolution(token0Decimals).div(priceB))) {
+      maxWithdrawB0 = maxWithdrawB0
+        .setResolution(token0Decimals)
```

Comment:
You shouldn't be setting the resolution here. Set the resolution before the div on line 312 instead.